### DISCUSSION TITLE
Updating to numpy 2.0 results to crashes

### DISCUSSION THREAD

**Discussion by MagiRomanya** [pybind/pybind11#5218] [Q&A]
After updating my system, I've been experiencing crashes in previously functional code. Here is a simplified version of the code:
- C++ side:
```c++
#include <pybind11/pybind11.h>
#include <pybind11/eigen.h>
#include <Eigen/Core>

struct MyTest {
    Eigen::Vector3f vector;
};

namespace py = pybind11;
PYBIND11_MODULE(MyModule, m) {
    py::class_<MyTest>(m, "MyTest")
        .def(py::init())
        .def_readwrite("vector", &MyTest::vector)
        ;
}
```
- Python side:
```python
import MyModule
test = MyModule.MyTest()
test.vector = [0.0, 0.0, 0.0]  # Segfaults here
# test.vector = np.array([0.0, 0.0, 0.0])  # This also segfaults
```
In my tests this only crashes with numpy 2.0 and not with 1.26.0. Is this is a bug in pybind11 or that I just programmed this naively expecting an implicit conversion from python list to np.array to Eigen::Matrix?

**Comment by atuleu** [comment]
I think it may be related to v2.12.0 of pybind that enabled numpy 2 support.

**Reply by atuleu** [reply]
Upgrading my project to use pybind 2.13.6 from ubuntu 22.04 2.9.2 solved this issue.

**Comment by atuleu** [comment]
I think this may be related to your version of pybind11. Numpy 2 support only exixst from version 2.12 onward. Unfortunately on some system, an older version is packaged by default. In my case, Ubuntu 22.04, comes only with 2.9.2, without numpy 2 support. Using FetchContent, to enforce pybind11 2.13.6 solved this issue for me.
